executor/builder.go

```diff
@@ -1010,16 +1012,20 @@ func (b *executorBuilder) constructIndexRanges(is *plan.PhysicalIndexScan) (newR
 }
 
 func (b *executorBuilder) buildIndexLookUpJoin(v *plan.PhysicalIndexJoin) Executor {
-	batchSize := 1
-	if !v.KeepOrder {
-		batchSize = b.ctx.GetSessionVars().IndexJoinBatchSize
-	}
-
 	outerExec := b.build(v.Children()[v.OuterIndex])
 	if b.err != nil {
 		b.err = errors.Trace(b.err)
 		return nil
 	}
+	if outerExec.supportChunk() {
+		// All inner data reader supports chunk(TableReader, IndexReader, IndexLookUpReadfer),
+		// we only need to check outer.
+		return b.buildNewIndexLookUpJoin(v, outerExec)
+	}
+	batchSize := 1
+	if !v.KeepOrder {
+		batchSize = b.ctx.GetSessionVars().IndexJoinBatchSize
```

Comment:
if `KeepOrder` is false, we can produce unordered joined rows during `NextChunk`, i.e the joined rows are allowed to not to be ordered by join keys.

Reply:
But make the join unordered doesn't make it faster for the new index look up join.

Reply:
if keep order is false, main thread no need to wait for the current task to be done, it can chose any tasks in `e.resultCh`

Reply:
We don't know which one is done until we wait for it.